### ISSUE TITLE
Regression Visualizer Improvements for Users with Red-Green Colorblindness

### ISSUE THREAD

**Issue by guindo** [DistrictDataLabs/yellowbrick#1134]
how to change markers of the residuals plot and prediction plot in order to have different markers for training set and test set. For users with daltonism, different markers sound be a better option intead of having different colors.
And also is there any alternative to have different color for best fit and identity in Prediction error plot instead of having same color and different opacity.

![residuals](https://user-images.githubusercontent.com/16453089/102528206-3408b980-40d9-11eb-9066-64343ac7e188.png)

**Comment by rebeccabilbro** [comment]
Hello @guindo and thank you for checking out Yellowbrick, and thank you especially for reaching out with this request regarding accessibility! I have to admit that, while I am familiar with colorblindness, I had not heard the term daltonism before! From some preliminary research, it appears that this describes red/green color blindness and generally makes it difficult to differentiate two colors that differ primarily in their amount of red; is that correct? 

We want to make Yellowbrick as accessible as we can &mdash; to that end, would you point me to some examples of visualizations that you feel are doing a good job to support readers with red-green colorblindess, so that I can see some good examples? Or, if there are not many good examples out there, might you be willing to team up with us to work on this? We are a small team of unpaid volunteers and would definitely appreciate the assistance!

In the meantime, to assist users with colorblindness, our Yellowbrick `Visualizers` do support [color customization](https://www.scikit-yb.org/en/develop/faq.html#how-can-i-change-the-color-of-a-yellowbrick-plot), so if you wanted to change the default colors for `ResidualsPlot`, you can do so with the `train_color` and `test_color` parameters, eg:

```python
from sklearn.linear_model import Ridge
from yellowbrick.datasets import load_concrete
from yellowbrick.regressor import ResidualsPlot
from sklearn.model_selection import train_test_split


X, y = load_concrete()
X_train, X_test, y_train, y_test = train_test_split(
    X, 
    y, 
    test_size=0.2, 
    random_state=38
)
model = Ridge()
visualizer = ResidualsPlot(
    model, 
    train_color="goldenrod", 
    test_color="darkblue"
)

visualizer.fit(X_train, y_train) 
visualizer.score(X_test, y_test)
visualizer.show()
```
![goldenrod_darkblue](https://user-images.githubusercontent.com/8760385/102658726-23daf380-4146-11eb-9d38-c0767ddd2169.png)

Do you know of a pair of colors that are particularly easy to differentiate (such as the above? or perhaps [these?](https://gist.github.com/thriveth/8560036)) for those with red-green colorblindness? Your suggestion about changing the markers is also interesting; could you direct me to some examples of scatterplots that use  markers in a way that you find particularly helpful for differentiation? I expect that the density of points might also be somewhat important; can you think of any other considerations?

It is also possible to change the color of the `identify` and `best_fit` lines in the `PredictionError` plot, using the `line_color` argument:

```python
from sklearn.linear_model import Lasso
from yellowbrick.datasets import load_concrete
from yellowbrick.regressor import PredictionError
from sklearn.model_selection import train_test_split

X, y = load_concrete()
X_train, X_test, y_train, y_test = train_test_split(
    X, 
    y, 
    test_size=0.2, 
    random_state=38
)model = Lasso()
visualizer = PredictionError(
    model, 
    line_color="midnightblue"
)

visualizer.fit(X_train, y_train) 
visualizer.score(X_test, y_test)
visualizer.show()
```
![midnightblue](https://user-images.githubusercontent.com/8760385/102660012-5c7bcc80-4148-11eb-9665-9bde0655926e.png)

Currently the `line_color` parameter changes both `identity` and `best_fit`, but it would not be too tricky to change that code by updating the [visual arguments](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/regressor/prediction_error.py#L130-L134) and how they are consumed. Can you suggest a pair of colors that are easily red-green differentiable and which you feel might make for better default values?

Thank you in advance for your help and for raising this important accessibility issue!

**Comment by guindo** [comment]
Thank you for your feedback. Daltonism or Colorblindness are same and both is a genetic defect that causes difficulty in distinguishing colors. Many people have some degree of inability to distinguish colors, but still do not know.
My suggestion is to offer users to change markers when visualizing .For example, the attached figure is made using Matplotlib you can see we have color but we have also different markers.
The reason why using marker first point is colorblindness, the second point imagine I download a scientific paper and print it out using black and white. I will have difficulty to understand the figure.
Thank you



![thumbnail](https://user-images.githubusercontent.com/16453089/102695458-80cbcc00-4262-11eb-95e6-8f4f384bdfc1.jpg)
